src/users/controllers/CreateUserController.js

```diff
@@ -0,0 +1,32 @@
+import { Prisma } from '@prisma/client';
+import { prismaClient } from '../../../database/prismaClient.js';
+import { encryptPassword } from  '../utils/encrypt.js';
+
+export class CreateUserController {
+    async handle(request, response) {
+        const {name, email, password} = request.body;
+        const hashPassword =  await encryptPassword(password)
+        try {
+            const user = await prismaClient.user.create({
+                data: {
+                    name,
+                    email,
+                    password: hashPassword
+                }
+            })
+
+            return response.json({message:`Usuario criado com sucesso - ID: ${user.id}`})
+          
+        } catch (e) {
+            console.log(e);
+            if(e instanceof Prisma.PrismaClientKnownRequestError) {
```

Comment:
aqui o teu erro, você pode passar essa logica pra um [middleware](https://expressjs.com/en/guide/using-.html) assim não precisa ficar implementando nas demais rotas que vai usar

Reply:
Hmm, eu vi como boa prática realmente criar Middlewares padrão de erro, já adicionei no meu To Do